carbonmark-api/src/routes/projects/get.ts

```diff
@@ -52,280 +54,93 @@ const schema = {
-          POOL_INFO.nct.defaultProjectTokenAddress.toLowerCase() ===
-          project.tokenAddress.toLowerCase();
-        const priceKey = isDefault ? "defaultPrice" : "selectiveRedeemPrice";
-        uniqueValues.push(poolPrices.nct[priceKey]);
-      }
-      if (parseFloat(project.balanceBCT) >= 1) {
-        const isDefault =
-          POOL_INFO.bct.defaultProjectTokenAddress.toLowerCase() ===
-          project.tokenAddress.toLowerCase();
-        const priceKey = isDefault ? "defaultPrice" : "selectiveRedeemPrice";
-        uniqueValues.push(poolPrices.bct[priceKey]);
-      }
-
-      const country = project.country.length ? { id: project.country } : null;
 
-      const cmsData = findProjectWithRegistryIdAndRegistry(
+    // ----- Pool Listings ----- //
+    const offsetProjects = offsetData.carbonOffsets.map((offset) => {
+      const cmsProject = findProjectWithRegistryIdAndRegistry(
         projectsCmsData,
-        project.projectID.split("-")[1],
-        project.projectID.split("-")[0]
+        offset.projectID.split("-")[1], //e.g 1120
+        offset.projectID.split("-")[0] //e.g VCS
       );
 
-      const singleProject = {
-        id: project.id,
-        isPoolProject: true,
-        description: cmsData ? cmsData.description : undefined,
-        short_description: cmsData?.projectContent
-          ? cmsData.projectContent.shortDescription
-          : undefined,
-        key: project.projectID,
-        projectID: project.projectID.split("-")[1],
-        name: cmsData ? cmsData.name : project.name,
-        methodologies: cmsData ? cmsData.methodologies : [],
-        vintage: project.vintageYear,
-        projectAddress: project.tokenAddress,
-        registry: project.projectID.split("-")[0],
-        updatedAt: project.lastUpdate,
-        category: {
-          id: project.methodologyCategory,
-        },
-        country: country,
-        price: uniqueValues.length
-          ? uniqueValues.reduce((a, b) =>
-              a.length < b.length ? a : a.length === b.length && a < b ? a : b
-            )
-          : "0",
-        activities: null,
-        listings: null,
-      };
+      // Find the lowest price
+      // @todo change to number[]
+      const tokenPrices = getOffsetTokenPrices(offset, poolPrices);
+      const lowestPrice = tokenPrices.sort().at(0);
 
-      return singleProject;
+      return composeOffsetProject(cmsProject, offset, lowestPrice);
     });
 
-    const filteredItems = projects
-      .concat(pooledProjects)
-      /**
-       * This is where we run in to trouble because a project might exist in both MARKETPLACE and pooledProjects but only contain pricing in the latter
-       * Which means it won't be displayed if it's price is zero but is present in MARKETPLACE
-       */
-      .filter((project) => Number(project?.price) !== 0);
+    const filteredProjects = concat(projects, offsetProjects).filter(
+      ({ price }) => notNil(price) && parseFloat(price) > 0
+    );
```

Comment:
Does this need to account if there is the same project in projects and offsetProjects with different marketplace and pool pricing?  The answer might be no as it looks like there is a isPoolProject bool in the type which I'm guessing is used to id the project later on but just a double-check

Reply:
Yea it's a good question. My understanding is that we will need to remove duplicates at some point. For now the marketplace projects are not displayed in Carbonmark anyway (because of the isActive filter) but that will change in the near future. 

I didn't want to change how the response is built for this PR to make sure I didnt break anything, rather just to clean up the code itself. Does that make sense?

Reply:
Sounds good